lib/fastclick.js

```diff
@@ -536,14 +531,14 @@
 	 */
 	FastClick.prototype.onTouchEnd = function(event) {
 		var forElement, trackingClickStart, targetTagName, scrollParent, touch, touchEndTime, targetElement = this.targetElement;
-
-		if (deviceIsIOS11_3) {
-			// Do not use event.timeStamp
-			// iOS 11.3 returns negative values for event.timeStamp after pausing/resuming a Cordova application
+	
+		if (this.isTrackingClickStartFromEvent) {
+			touchEndTime = event.timeStamp;
+		} else {
+			// iOS (at least 11.3 and 11.4 beta) returns negative values for event.timeStamp after pausing/resuming a Cordova application
 			// and supposedly for mobile Safari as well
+			// https://github.com/ftlabs/fastclick/issues/549
 			touchEndTime = (new Date()).getTime();
-		} else {
-			touchEndTime = event.timeStamp;
 		}
 
```

Comment:
Is it possible for `event.timeStamp` to be positive for the touchstart event but negative for the touchend event? In which case, it would be better to only use `(new Date()).getTime()` instead of the event.timeStamp.

Otherwise, the comparisons below would cause the touchend to exit early:

```
 		// Prevent phantom clicks on fast double-tap (issue #36)
 		if ((touchEndTime - this.lastClickTime) < this.tapDelay) {
```